File: Java/src/main/java/com/anexsys/recruitment/challenge/Game.java
package com.anexsys.recruitment.challenge;

import java.util.concurrent.CompletableFuture;

interface Game {
    interface GameComplete {
        Player winner();
    }
    Game shuffle();
    Game assign(Player[] players);
    Game deal();
    CompletableFuture<GameComplete> start();
}

File: CSharp/CSharp/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharp
{
    interface Game
    {
        Game Shuffle();
        Game Assign(Player[] players);
        Game Deal();

        Task<Player> Start();

        Task<Player> Winner();
    }
}
